response_operations_ui/templates/collection-exercise.html

```diff
@@ -141,18 +176,32 @@ <h2 class="venus u-pt-m">Dates</h2>
               <td class="table--cell tbl-ce-value ce-event-date" name="third-reminder-date">
                 {{events.reminder3.day}} <em>{{events.reminder3.date}}</em> {{events.reminder3.time}}
               </td>
+              {% if events.reminder3.is_in_future and config.EDIT_EVENT_DATES_ENABLED %}
+              <td class="table--cell tbl-ce-">
+                <form action="{{ url_for('collection_exercise_bp.update_event_date', short_name=survey.shortName, period=ce.exerciseRef, tag='reminder3')}}" method="get">
+                  <a class="edit-icon" href=# id="edit-event-date-reminder3" onclick="this.parentNode.submit()"></a>
+                </form>
+              </td>
+              {% endif %}
             </tr>
             {% endif %}
 
             <tr class="table--row">
               <td class="table--cell tbl-ce-label">
                 Exercise end
               </td>
+              {% if events.exercise_end %}
               <td class="table--cell tbl-ce-value ce-event-date" name="exercise-end-date">
-                {% if events.exercise_end %}
                 {{events.exercise_end.day}} <em>{{events.exercise_end.date}}</em> {{events.exercise_end.time}}
-                {% endif %}
               </td>
+              {% if not ce.state == 'Live' and not ce.state == 'Ready for live' and events.exercise_end.is_in_future and config.EDIT_EVENT_DATES_ENABLED %}
```

Comment:
This seems a bit long to me.  Having 4 things 'and'ed together seems excessive in a template.....but I'm not sure if it's enough to block it before wanting it merged in

Reply:
Yeah I dislike having so much logic in the templates as well. There's probably a wider task around cleaning up how we're using jinja2 that that could maybe be a part of